docs/source/reference/envvars.rst

```diff
@@ -299,7 +299,16 @@ Compilation options
 
 .. envvar:: NUMBA_OPT
 
-   The optimization level; this option is passed straight to LLVM.
+   The optimization level; typically this option is passed straight to LLVM. It
+   may take one of the values ``0``, ``1``, ``2`` or ``3`` which correspond
+   approximately to the ``-O{value}`` flag found in many command line
+   compilation tools. The value ``max`` is also supported, this is Numba
+   specific, it has the effect of running with the optimization level set at
```

Comment:
Can we phrase it so that what `max` does exactly may change but it currently does it the certain way? We don't need to promise that it will always run O3 twice. It may change as we discover other ways. The key thing is that its effect is to sacrifice compile time for better runtime.

Reply:
That's a good point, I'll rephrase prose that describe this effect to contain something more general.

Reply:
Implemented in c91b102.